src/habit_server/db_manager.py

```diff
@@ -0,0 +1,163 @@
+"""Database manager for the habit server."""
```

Comment:
Merge in Jared's changes from his branch after 11/30, assume his changes take priority in conflicts.